<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>BayesSpace analysis of SCC dataset (Ji et al., 2020) • BayesSpace</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="BayesSpace analysis of SCC dataset (Ji et al., 2020)">
<meta property="og:description" content="BayesSpace">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BayesSpace</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home"></span>
     
  </a>
</li>
<li>
  <a href="../articles/BayesSpace.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Examples
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/maynard_DLPFC.html">DLPFC (Maynard et al., 2020)</a>
    </li>
    <li>
      <a href="../articles/ji_SCC.html">SCC (Ji et al., 2020)</a>
    </li>
    <li>
      <a href="../articles/thrane_melanoma.html">Melanoma (Thrane et al., 2018)</a>
    </li>
    <li>
      <a href="../articles/joint_clustering.html">Joint clustering</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/edward130603/BayesSpace">
    <span class="fas fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="ji_SCC_files/header-attrs-2.6/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>BayesSpace analysis of SCC dataset (Ji et al., 2020)</h1>
                        <h4 class="author">Edward Zhao, Matt Stone, Xing Ren, and Raphael Gottardo</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/edward130603/BayesSpace/blob/master/vignettes/ji_SCC.Rmd"><code>vignettes/ji_SCC.Rmd</code></a></small>
      <div class="hidden name"><code>ji_SCC.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BayesSpace</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com">patchwork</a></span><span class="op">)</span></code></pre></div>
<div id="overview" class="section level2">
<h2 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h2>
<p>Here we present our re-analysis of one of the squamous cell carcinoma (SCC) samples originally reported by <a href="https://www.sciencedirect.com/science/article/pii/S0092867420306723?via%3Dihub">Ji et al. (2020)</a>. These data were obtained from GEO (<a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE144239">accession GSE144239</a>); we re-analyze the sample from patient 4, which had greater sequencing depth than the sample from patient 6.</p>
</div>
<div id="processing-the-data" class="section level2">
<h2 class="hasAnchor">
<a href="#processing-the-data" class="anchor"></a>Processing the data</h2>
<p>A cleaned <code>SingleCellExperiment</code> object containing the dataset is available through BayesSpace (details on data cleaning are available on <a href="https://github.com/msto/spatial-datasets/blob/master/snakefiles/2020_ji_squamous-cell-carcinoma.snake">Github</a>). We preprocessed the data by performing PCA on the top 2,000 HVGs.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">scc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getRDS.html">getRDS</a></span><span class="op">(</span><span class="st">"2020_ji_squamous-cell-carcinoma"</span>, <span class="st">"P4_rep1"</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">101</span><span class="op">)</span>
<span class="va">dec</span> <span class="op">&lt;-</span> <span class="fu">scran</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scran/man/modelGeneVar.html">modelGeneVar</a></span><span class="op">(</span><span class="va">scc</span><span class="op">)</span>
<span class="va">top</span> <span class="op">&lt;-</span> <span class="fu">scran</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scran/man/getTopHVGs.html">getTopHVGs</a></span><span class="op">(</span><span class="va">dec</span>, n <span class="op">=</span> <span class="fl">2000</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">102</span><span class="op">)</span>
<span class="va">scc</span> <span class="op">&lt;-</span> <span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/runPCA.html">runPCA</a></span><span class="op">(</span><span class="va">scc</span>, subset_row<span class="op">=</span><span class="va">top</span><span class="op">)</span>

<span class="co">## Add BayesSpace metadata</span>
<span class="va">scc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/spatialPreprocess.html">spatialPreprocess</a></span><span class="op">(</span><span class="va">scc</span>, platform<span class="op">=</span><span class="st">"Visium"</span>, skip.PCA<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
</div>
<div id="clustering-with-bayesspace" class="section level2">
<h2 class="hasAnchor">
<a href="#clustering-with-bayesspace" class="anchor"></a>Clustering with BayesSpace</h2>
<p>Unlike our analyses of the melanoma (Thrane et al. 2018) and dorsolateral prefrontal cortex (Maynard, Collado-Torres, et al. 2020) samples, this sample did not have a strong biological prior for the expected number of clusters. After evaluating the average model pseudo-log-likelihood over the first 1000 iterations (excluding a 100-iteration burn-in), we selected <code>q=12</code> as a reasonable elbow and choice of number of clusters.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">scc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/qTune.html">qTune</a></span><span class="op">(</span><span class="va">scc</span>, qs<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">20</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="../reference/qTune.html">qPlot</a></span><span class="op">(</span><span class="va">scc</span><span class="op">)</span></code></pre></div>
<p><img src="figures/ji_SCC-qTune-1.png" title="plot of chunk qTune" alt="plot of chunk qTune" width="100%"></p>
<p>We clustered the first 15 principal components, specifying 12 clusters as described above, and ran the MCMC algorithm for 10,000 iterations. We set our smoothing parameter <code>gamma</code> to 3, which we generally suggest for Visium datasets. We also computed an enhanced resolution clustering using the same parameters and 200,000 iterations.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">q</span> <span class="op">&lt;-</span> <span class="fl">12</span>  <span class="co"># Number of clusters</span>
<span class="va">d</span> <span class="op">&lt;-</span> <span class="fl">15</span>  <span class="co"># Number of PCs</span>

<span class="co">## Here we run mclust externally so the random seeding is consistent with </span>
<span class="co">## original analyses</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://mclust-org.github.io/mclust/">mclust</a></span><span class="op">)</span>
<span class="va">Y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html">reducedDim</a></span><span class="op">(</span><span class="va">scc</span>, <span class="st">"PCA"</span><span class="op">)</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span><span class="op">(</span><span class="va">d</span><span class="op">)</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">101</span><span class="op">)</span>
<span class="va">init</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mclust-org.github.io/mclust/reference/Mclust.html">Mclust</a></span><span class="op">(</span><span class="va">Y</span>, <span class="va">q</span>, <span class="st">"EEE"</span>, verbose<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span><span class="op">$</span><span class="va">classification</span>

<span class="co">## Run BayesSpace clustering</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">100</span><span class="op">)</span>
<span class="va">scc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/spatialCluster.html">spatialCluster</a></span><span class="op">(</span><span class="va">scc</span>, q<span class="op">=</span><span class="va">q</span>, d<span class="op">=</span><span class="va">d</span>, platform<span class="op">=</span><span class="st">'Visium'</span>, init<span class="op">=</span><span class="va">init</span>,
                      nrep<span class="op">=</span><span class="fl">10000</span>, gamma<span class="op">=</span><span class="fl">3</span><span class="op">)</span>

<span class="co">## Run BayesSpace enhanced clustering</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">100</span><span class="op">)</span>
<span class="va">scc.enhanced</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/spatialEnhance.html">spatialEnhance</a></span><span class="op">(</span><span class="va">scc</span>, q<span class="op">=</span><span class="va">q</span>, d<span class="op">=</span><span class="va">d</span>, platform<span class="op">=</span><span class="st">"Visium"</span>,
                               nrep<span class="op">=</span><span class="fl">200000</span>, gamma<span class="op">=</span><span class="fl">3</span>, verbose<span class="op">=</span><span class="cn">TRUE</span>,
                               jitter_scale<span class="op">=</span><span class="fl">5.5</span>, jitter_prior<span class="op">=</span><span class="fl">0.3</span>,
                               save.chain<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>We compared the two clusterings using <code><a href="../reference/clusterPlot.html">clusterPlot()</a></code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">palette</span> <span class="op">&lt;-</span> <span class="fu">RColorBrewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html">brewer.pal</a></span><span class="op">(</span><span class="va">q</span>, <span class="st">"Paired"</span><span class="op">)</span>

<span class="va">spot.plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/clusterPlot.html">clusterPlot</a></span><span class="op">(</span><span class="va">scc</span>, palette<span class="op">=</span><span class="va">palette</span>, size<span class="op">=</span><span class="fl">0.05</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title<span class="op">=</span><span class="st">"Spot-level clustering"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html">guides</a></span><span class="op">(</span>fill<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>

<span class="va">enhanced.plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/clusterPlot.html">clusterPlot</a></span><span class="op">(</span><span class="va">scc.enhanced</span>, palette<span class="op">=</span><span class="va">palette</span>, size<span class="op">=</span><span class="fl">0.05</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title<span class="op">=</span><span class="st">"Enhanced clustering"</span><span class="op">)</span>

<span class="va">spot.plot</span> <span class="op">+</span> <span class="va">enhanced.plot</span></code></pre></div>
<p><img src="figures/ji_SCC-plot.clusters-1.png" title="plot of chunk plot.clusters" alt="plot of chunk plot.clusters" width="100%"></p>
</div>
<div id="enhancing-feature-expression" class="section level2">
<h2 class="hasAnchor">
<a href="#enhancing-feature-expression" class="anchor"></a>Enhancing feature expression</h2>
<p>Next we examined the expression of cell-type-specific marker genes reported by Ji et al. at the spot level and after enhancement. We predicted the marker gene at enhanced subspot resolution using <code>xgboost</code>, tuning the selection of the <code>nrounds</code> parameter with two thirds of the spots. (See documentation of <code><a href="../reference/enhanceFeatures.html">enhanceFeatures()</a></code> for details.)</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">markers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">markers</span><span class="op">[[</span><span class="st">"Basal keratinocyte"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"KRT5"</span>, <span class="st">"KRT14"</span><span class="op">)</span>
<span class="va">markers</span><span class="op">[[</span><span class="st">"Suprabasal keratinocyte"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"KRT1"</span>, <span class="st">"KRT10"</span><span class="op">)</span>
<span class="va">markers</span><span class="op">[[</span><span class="st">"Myeloid"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"LYZ"</span><span class="op">)</span>
<span class="va">markers</span><span class="op">[[</span><span class="st">"T-cell"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"CD2"</span>, <span class="st">"CD3D"</span>, <span class="st">"CD3E"</span>, <span class="st">"CD3G"</span>, <span class="st">"CD7"</span><span class="op">)</span>
<span class="va">markers</span><span class="op">[[</span><span class="st">"Melanocyte"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"MLANA"</span>, <span class="st">"DCT"</span>, <span class="st">"PMEL"</span><span class="op">)</span>
<span class="va">markers</span><span class="op">[[</span><span class="st">"Endothelial"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"TFF3"</span>, <span class="st">"CLDN5"</span>, <span class="st">"VWF"</span><span class="op">)</span>

<span class="va">scc.enhanced</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/enhanceFeatures.html">enhanceFeatures</a></span><span class="op">(</span><span class="va">scc.enhanced</span>, <span class="va">scc</span>,
                                model<span class="op">=</span><span class="st">"xgboost"</span>,
                                feature_names<span class="op">=</span><span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/reduce.html">reduce</a></span><span class="op">(</span><span class="va">markers</span>, <span class="va">c</span><span class="op">)</span>,
                                nrounds<span class="op">=</span><span class="fl">0</span><span class="op">)</span></code></pre></div>
<p>We aggregated the expression of marker genes within each cell type by summing their log-normalized expression.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sum_counts</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">sce</span>, <span class="va">features</span><span class="op">)</span> <span class="op">{</span>
  <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">features</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">1</span><span class="op">)</span> <span class="op">{</span>
    <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">colSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/assays.html">logcounts</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">[</span><span class="va">features</span>, <span class="op">]</span><span class="op">)</span>
  <span class="op">}</span> <span class="kw">else</span> <span class="op">{</span>
    <span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/assays.html">logcounts</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">[</span><span class="va">features</span>, <span class="op">]</span>
  <span class="op">}</span>
<span class="op">}</span>

<span class="va">spot_expr</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map</a></span><span class="op">(</span><span class="va">markers</span>, <span class="kw">function</span><span class="op">(</span><span class="va">xs</span><span class="op">)</span> <span class="fu">sum_counts</span><span class="op">(</span><span class="va">scc</span>, <span class="va">xs</span><span class="op">)</span><span class="op">)</span>
<span class="va">enhanced_expr</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map</a></span><span class="op">(</span><span class="va">markers</span>, <span class="kw">function</span><span class="op">(</span><span class="va">xs</span><span class="op">)</span> <span class="fu">sum_counts</span><span class="op">(</span><span class="va">scc.enhanced</span>, <span class="va">xs</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>And we plotted the spatial expression of each cell type’s markers using <code><a href="../reference/featurePlot.html">featurePlot()</a></code>, at the spot-level:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plot_expression</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">sce</span>, <span class="va">expr</span>, <span class="va">name</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="../reference/featurePlot.html">featurePlot</a></span><span class="op">(</span><span class="va">sce</span>, <span class="va">expr</span>, color<span class="op">=</span><span class="cn">NA</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">viridis</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/viridis/man/scale_viridis.html">scale_fill_viridis</a></span><span class="op">(</span>option<span class="op">=</span><span class="st">"A"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title<span class="op">=</span><span class="va">name</span>, fill<span class="op">=</span><span class="st">"Log-normalized\nexpression"</span><span class="op">)</span>
<span class="op">}</span>

<span class="va">spot_plots</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/imap.html">imap</a></span><span class="op">(</span><span class="va">spot_expr</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">y</span><span class="op">)</span> <span class="fu">plot_expression</span><span class="op">(</span><span class="va">scc</span>, <span class="va">x</span>, <span class="va">y</span><span class="op">)</span><span class="op">)</span>
<span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html">wrap_plots</a></span><span class="op">(</span><span class="va">spot_plots</span>, ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span></code></pre></div>
<p><img src="figures/ji_SCC-spot.expression-1.png" title="plot of chunk spot.expression" alt="plot of chunk spot.expression" width="100%"></p>
<p>And at enhanced subspot resolution:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">enhanced_plots</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/imap.html">imap</a></span><span class="op">(</span><span class="va">enhanced_expr</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">y</span><span class="op">)</span> <span class="fu">plot_expression</span><span class="op">(</span><span class="va">scc.enhanced</span>, <span class="va">x</span>, <span class="va">y</span><span class="op">)</span><span class="op">)</span>
<span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html">wrap_plots</a></span><span class="op">(</span><span class="va">enhanced_plots</span>, ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span></code></pre></div>
<p><img src="figures/ji_SCC-enhanced.expression-1.png" title="plot of chunk enhanced.expression" alt="plot of chunk enhanced.expression" width="100%"></p>
</div>
<div id="differential-expression-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#differential-expression-analysis" class="anchor"></a>Differential expression analysis</h2>
<p>Finally, we performed a differential expression analysis between clusters of interest. We limited our analysis to the top 2,000 highly variable genes, and we first imputed the expression of these HVGs at enhanced resolution.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Using the same 2,000 HVGs previously computed for PCA</span>
<span class="va">scc.enhanced</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/enhanceFeatures.html">enhanceFeatures</a></span><span class="op">(</span><span class="va">scc.enhanced</span>, <span class="va">scc</span>, 
                                model<span class="op">=</span><span class="st">"xgboost"</span>,
                                feature_names<span class="op">=</span><span class="va">top</span>,
                                nrounds<span class="op">=</span><span class="fl">0</span><span class="op">)</span></code></pre></div>
<p>Next, we applied a standard Seurat differential expression analysis workflow to the enhanced resolution.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>
<span class="va">seurat_cluster_DE</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">sce</span>, <span class="va">clusters</span><span class="op">=</span><span class="cn">NULL</span>, <span class="va">n_markers</span><span class="op">=</span><span class="fl">6</span><span class="op">)</span> <span class="op">{</span>
  <span class="co">## Convert SCE to seurat object and use BayesSpace cluster as identifier</span>
  <span class="va">seurat</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/CreateSeuratObject.html">CreateSeuratObject</a></span><span class="op">(</span>counts<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/assays.html">logcounts</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span>,
                                       assay<span class="op">=</span><span class="st">'Spatial'</span>,
                                       meta.data<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="va">seurat</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/Idents.html">SetIdent</a></span><span class="op">(</span><span class="va">seurat</span>, value <span class="op">=</span> <span class="st">"spatial.cluster"</span><span class="op">)</span>
  
  <span class="co">## Subset to specified clusters</span>
  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html">is.null</a></span><span class="op">(</span><span class="va">clusters</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">seurat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">seurat</span>, <span class="va">spatial.cluster</span> <span class="op">%in%</span> <span class="va">clusters</span><span class="op">)</span>
    <span class="va">palette</span> <span class="op">&lt;-</span> <span class="va">palette</span><span class="op">[</span><span class="va">clusters</span><span class="op">]</span>
  <span class="op">}</span>
  
  <span class="co">## Scale data</span>
  <span class="va">seurat</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">Spatial</span><span class="op">@</span><span class="va">scale.data</span> <span class="op">&lt;-</span>
    <span class="va">seurat</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">Spatial</span><span class="op">@</span><span class="va">data</span> <span class="op">%&gt;%</span> <span class="va">as.matrix</span> <span class="op">%&gt;%</span> <span class="va">t</span> <span class="op">%&gt;%</span> <span class="va">scale</span> <span class="op">%&gt;%</span> <span class="va">t</span>
  
  <span class="co">## Select top n markers from each cluster (by log fold change)</span>
  <span class="va">top_markers</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/FindAllMarkers.html">FindAllMarkers</a></span><span class="op">(</span><span class="va">seurat</span>, assay<span class="op">=</span><span class="st">'Spatial'</span>, slot<span class="op">=</span><span class="st">'data'</span>,
                                        group.by<span class="op">=</span><span class="st">'spatial.cluster'</span>,
                                        logfc.threshold<span class="op">=</span><span class="fl">1</span>, only.pos<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span> <span class="op">%&gt;%</span> 
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">cluster</span><span class="op">)</span> <span class="op">%&gt;%</span> 
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/top_n.html">top_n</a></span><span class="op">(</span><span class="va">n_markers</span>, <span class="va">avg_logFC</span><span class="op">)</span>
  
  <span class="co">## Plot expression of markers</span>
  <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/DoHeatmap.html">DoHeatmap</a></span><span class="op">(</span><span class="va">seurat</span>, features <span class="op">=</span> <span class="va">top_markers</span><span class="op">$</span><span class="va">gene</span>, slot<span class="op">=</span><span class="st">'scale.data'</span>,
                    group.by <span class="op">=</span> <span class="st">"spatial.cluster"</span>, group.colors<span class="op">=</span><span class="va">palette</span>, 
                    angle<span class="op">=</span><span class="fl">0</span>, size<span class="op">=</span><span class="fl">4</span>, label <span class="op">=</span> <span class="cn">FALSE</span>, raster<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html">guides</a></span><span class="op">(</span>col <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
<p>We highlight the different expression between clusters 4 and 8, which both demonstrate enriched immune cell expression but display substantial differences in expression of immunoglobulin genes and genes regulated by interferons.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">seurat_cluster_DE</span><span class="op">(</span><span class="va">scc.enhanced</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">4</span>, <span class="fl">8</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="figures/ji_SCC-DE.myeloid-1.png" title="plot of chunk DE.myeloid" alt="plot of chunk DE.myeloid" width="100%"></p>
<p>We additionally highlight the heterogeneity within the tumor revealed by differential expression of the tumor clusters.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">seurat_cluster_DE</span><span class="op">(</span><span class="va">scc.enhanced</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">6</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="figures/ji_SCC-DE.tumor-1.png" title="plot of chunk DE.tumor" alt="plot of chunk DE.tumor" width="100%"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Edward Zhao, Matt Stone.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
