<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>BayesSpace analysis of melanoma dataset (Thrane et al., 2018) â€¢ BayesSpace</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="BayesSpace analysis of melanoma dataset (Thrane et al., 2018)">
<meta property="og:description" content="BayesSpace">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BayesSpace</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li>
  <a href="../articles/BayesSpace.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Examples
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/maynard_DLPFC.html">DLPFC (Maynard et al., 2020)</a>
    </li>
    <li>
      <a href="../articles/ji_SCC.html">SCC (Ji et al., 2020)</a>
    </li>
    <li>
      <a href="../articles/thrane_melanoma.html">Melanoma (Thrane et al., 2018)</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/edward130603/BayesSpace">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="thrane_melanoma_files/header-attrs-2.3/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>BayesSpace analysis of melanoma dataset (Thrane et al., 2018)</h1>
                        <h4 class="author">Edward Zhao, Matt Stone, Xing Ren, and Raphael Gottardo</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/edward130603/BayesSpace/blob/master/vignettes/thrane_melanoma.Rmd"><code>vignettes/thrane_melanoma.Rmd</code></a></small>
      <div class="hidden name"><code>thrane_melanoma.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw">BayesSpace</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="http://ggplot2.tidyverse.org">ggplot2</a></span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="https://patchwork.data-imaginist.com">patchwork</a></span>)
</pre></div>
<div id="overview" class="section level2">
<h2 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h2>
<p>Here we present our re-analysis of one of the melanoma samples originally reported by <a href="https://cancerres.aacrjournals.org/content/78/20/5970.long">Thrane et al. (2018)</a>. These data were originally obtained through their <a href="https://www.spatialresearch.org/wp-content/uploads/2019/03/ST-Melanoma-Datasets_1.zip">website</a>. (NOTE: Since downloading this data, the Spatial Research website has gone offline. We will update this vignette with a working link to the original data when possible. Until then, the data remains available through our <code><a href="../reference/getRDS.html">getRDS()</a></code> funtion below.)</p>
</div>
<div id="processing-the-data" class="section level2">
<h2 class="hasAnchor">
<a href="#processing-the-data" class="anchor"></a>Processing the data</h2>
<p>A cleaned <code>SingleCellExperiment</code> object containing the dataset is available through BayesSpace. We preprocessed the data by performing PCA on the top 2,000 HVGs.</p>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="kw">melanoma</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getRDS.html">getRDS</a></span>(<span class="st">"2018_thrane_melanoma"</span>, <span class="st">"ST_mel1_rep2"</span>)

<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="fl">100</span>)
<span class="kw">dec</span> <span class="op">&lt;-</span> <span class="kw">scran</span>::<span class="fu"><a href="https://rdrr.io/pkg/scran/man/modelGeneVar.html">modelGeneVar</a></span>(<span class="kw">melanoma</span>)
<span class="kw">top</span> <span class="op">&lt;-</span> <span class="kw">scran</span>::<span class="fu"><a href="https://rdrr.io/pkg/scran/man/getTopHVGs.html">getTopHVGs</a></span>(<span class="kw">dec</span>, n = <span class="fl">2000</span>)

<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="fl">101</span>)
<span class="kw">melanoma</span> <span class="op">&lt;-</span> <span class="kw">scater</span>::<span class="fu"><a href="https://rdrr.io/pkg/scater/man/runPCA.html">runPCA</a></span>(<span class="kw">melanoma</span>, subset_row = <span class="kw">top</span>)

<span class="co">## Add BayesSpace metadata</span>
<span class="kw">melanoma</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/spatialPreprocess.html">spatialPreprocess</a></span>(<span class="kw">melanoma</span>, platform=<span class="st">"ST"</span>, skip.PCA=<span class="fl">TRUE</span>)

<span class="kw">q</span> <span class="op">&lt;-</span> <span class="fl">4</span>  <span class="co"># Number of clusters</span>
<span class="kw">d</span> <span class="op">&lt;-</span> <span class="fl">7</span>  <span class="co"># Number of PCs</span>
</pre></div>
</div>
<div id="clustering-with-bayesspace" class="section level2">
<h2 class="hasAnchor">
<a href="#clustering-with-bayesspace" class="anchor"></a>Clustering with BayesSpace</h2>
<p>We clustered the first seven principal components, specifying 4 clusters, and ran the MCMC algorithm for 50,000 iterations. We generally suggest setting the smoothing parameter gamma to 2 for ST experiments and to 3 for Visium data.</p>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="co">## Run BayesSpace clustering</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="fl">100</span>)
<span class="kw">melanoma</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/spatialCluster.html">spatialCluster</a></span>(<span class="kw">melanoma</span>, q=<span class="kw">q</span>, d=<span class="kw">d</span>, platform=<span class="st">'ST'</span>,
                           nrep=<span class="fl">50000</span>, gamma=<span class="fl">2</span>)

<span class="co">## View results</span>
<span class="kw">palette</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"purple"</span>, <span class="st">"red"</span>, <span class="st">"blue"</span>, <span class="st">"yellow"</span>, <span class="st">"darkblue"</span>)
<span class="fu"><a href="../reference/clusterPlot.html">clusterPlot</a></span>(<span class="kw">melanoma</span>, palette=<span class="kw">palette</span>, color=<span class="st">"black"</span>, size=<span class="fl">0.1</span>) <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span>(title=<span class="st">"BayesSpace"</span>)
</pre></div>
<p><img src="figures/melanoma-spatialCluster-1.png" title="plot of chunk spatialCluster" alt="plot of chunk spatialCluster" width="100%"></p>
</div>
<div id="enhancing-resolution-with-bayesspace" class="section level2">
<h2 class="hasAnchor">
<a href="#enhancing-resolution-with-bayesspace" class="anchor"></a>Enhancing resolution with BayesSpace</h2>
<p>Next, we clustered the dataset at enhanced resolution, using the same parameters and running the algorithm for 200,000 iterations.</p>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="fl">100</span>)
<span class="kw">melanoma.enhanced</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/spatialEnhance.html">spatialEnhance</a></span>(<span class="kw">melanoma</span>, q=<span class="kw">q</span>, d=<span class="kw">d</span>, platform=<span class="st">"ST"</span>,
                                    nrep=<span class="fl">200000</span>, gamma=<span class="fl">2</span>, 
                                    verbose=<span class="fl">TRUE</span>, save.chain=<span class="fl">TRUE</span>,
                                    jitter_scale=<span class="fl">3.5</span>, jitter_prior=<span class="fl">0.3</span>)

<span class="fu"><a href="../reference/clusterPlot.html">clusterPlot</a></span>(<span class="kw">melanoma.enhanced</span>, palette=<span class="kw">palette</span>, color=<span class="st">"black"</span>, size=<span class="fl">0.1</span>) <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span>(title=<span class="st">"Enhanced clustering"</span>)
</pre></div>
<p><img src="figures/melanoma-spatialEnhance-1.png" title="plot of chunk spatialEnhance" alt="plot of chunk spatialEnhance" width="100%"></p>
</div>
<div id="enhancement-of-marker-gene-expression" class="section level2">
<h2 class="hasAnchor">
<a href="#enhancement-of-marker-gene-expression" class="anchor"></a>Enhancement of marker gene expression</h2>
<p>We highlighted patterns of expression of cell type markers at both original spot and enhanced subspot resolution. First, we selected the below sets of marker genes for each of five cell types, and predicted their expression at subspot resolution using <code>xgboost</code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit">
<span class="kw">markers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>()
<span class="kw">markers</span>[[<span class="st">"Tumor"</span>]] <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"PMEL"</span>)
<span class="kw">markers</span>[[<span class="st">"Fibroblast"</span>]] <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"COL1A1"</span>)
<span class="kw">markers</span>[[<span class="st">"Macrophage"</span>]] <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"CD14"</span>, <span class="st">"FCGR1A"</span>, <span class="st">"FCGR1B"</span>)
<span class="kw">markers</span>[[<span class="st">"B-cell"</span>]] <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"CD19"</span>, <span class="st">"MS4A1"</span>)
<span class="kw">markers</span>[[<span class="st">"T-cell"</span>]] <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"CD2"</span>, <span class="st">"CD3D"</span>, <span class="st">"CD3E"</span>, <span class="st">"CD3G"</span>, <span class="st">"CD7"</span>)

<span class="kw">melanoma.enhanced</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/enhanceFeatures.html">enhanceFeatures</a></span>(<span class="kw">melanoma.enhanced</span>, <span class="kw">melanoma</span>,
                                     model=<span class="st">"xgboost"</span>,
                                     feature_names=<span class="kw">purrr</span>::<span class="fu"><a href="https://purrr.tidyverse.org/reference/reduce.html">reduce</a></span>(<span class="kw">markers</span>, <span class="kw">c</span>),
                                     nrounds=<span class="fl">0</span>)
</pre></div>
<p>Next, we aggregated the expression of marker genes within each cell type by summing their log-normalized expression.</p>
<div class="sourceCode" id="cb6"><pre class="downlit">
<span class="kw">sum_counts</span> <span class="op">&lt;-</span> <span class="fu">function</span>(<span class="kw">sce</span>, <span class="kw">features</span>) {
  <span class="co">if</span> (<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="kw">features</span>) <span class="op">&gt;</span> <span class="fl">1</span>) {
    <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">colSums</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/assays.html">logcounts</a></span>(<span class="kw">sce</span>)[<span class="kw">features</span>, ])
  } <span class="co">else</span> {
    <span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/assays.html">logcounts</a></span>(<span class="kw">sce</span>)[<span class="kw">features</span>, ]
  }
}

<span class="kw">spot_expr</span> <span class="op">&lt;-</span> <span class="kw">purrr</span>::<span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map</a></span>(<span class="kw">markers</span>, <span class="fu">function</span>(<span class="kw">xs</span>) <span class="fu">sum_counts</span>(<span class="kw">melanoma</span>, <span class="kw">xs</span>))
<span class="kw">enhanced_expr</span> <span class="op">&lt;-</span> <span class="kw">purrr</span>::<span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map</a></span>(<span class="kw">markers</span>, <span class="fu">function</span>(<span class="kw">xs</span>) <span class="fu">sum_counts</span>(<span class="kw">melanoma.enhanced</span>, <span class="kw">xs</span>))
</pre></div>
<p>And we plotted the spatial expression of each cell typeâ€™s markers at spot-level and enhanced subspot resolution using <code><a href="../reference/featurePlot.html">featurePlot()</a></code>:</p>
<div class="sourceCode" id="cb7"><pre class="downlit">
<span class="kw">plot_expression</span> <span class="op">&lt;-</span> <span class="fu">function</span>(<span class="kw">sce</span>, <span class="kw">expr</span>, <span class="kw">title</span>) {
  <span class="fu"><a href="../reference/featurePlot.html">featurePlot</a></span>(<span class="kw">sce</span>, <span class="kw">expr</span>, color=<span class="fl">NA</span>) <span class="op">+</span>
    <span class="kw">viridis</span>::<span class="fu"><a href="https://rdrr.io/pkg/viridis/man/scale_viridis.html">scale_fill_viridis</a></span>(option=<span class="st">"A"</span>) <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span>(title=<span class="kw">title</span>, fill=<span class="st">"Log-normalized\nexpression"</span>)
}

<span class="kw">plot_expression_comparison</span> <span class="op">&lt;-</span> <span class="fu">function</span>(<span class="kw">cell_type</span>) {
  <span class="kw">spot.plot</span> <span class="op">&lt;-</span> <span class="fu">plot_expression</span>(<span class="kw">melanoma</span>, 
                               <span class="kw">spot_expr</span>[[<span class="kw">cell_type</span>]], 
                               <span class="st">"Spot"</span>)
  <span class="kw">enhanced.plot</span> <span class="op">&lt;-</span> <span class="fu">plot_expression</span>(<span class="kw">melanoma.enhanced</span>,
                                   <span class="kw">enhanced_expr</span>[[<span class="kw">cell_type</span>]], 
                                   <span class="st">"Enhanced"</span>)
  
  (<span class="kw">spot.plot</span> <span class="op">+</span> <span class="kw">enhanced.plot</span>) <span class="op">+</span> 
    <span class="fu"><a href="https://patchwork.data-imaginist.com/reference/plot_annotation.html">plot_annotation</a></span>(title=<span class="kw">cell_type</span>,
                    theme=<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(plot.title=<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span>(size=<span class="fl">18</span>)))
}
</pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit">
<span class="fu">plot_expression_comparison</span>(<span class="st">"Tumor"</span>)
</pre></div>
<p><img src="figures/melanoma-plot.expression_tumor-1.png" title="plot of chunk plot.expression_tumor" alt="plot of chunk plot.expression_tumor" width="100%"></p>
<div class="sourceCode" id="cb9"><pre class="downlit">
<span class="fu">plot_expression_comparison</span>(<span class="st">"Fibroblast"</span>)
</pre></div>
<p><img src="figures/melanoma-plot.expression_fibro-1.png" title="plot of chunk plot.expression_fibro" alt="plot of chunk plot.expression_fibro" width="100%"></p>
<div class="sourceCode" id="cb10"><pre class="downlit">
<span class="fu">plot_expression_comparison</span>(<span class="st">"Macrophage"</span>)
</pre></div>
<p><img src="figures/melanoma-plot.expression_macro-1.png" title="plot of chunk plot.expression_macro" alt="plot of chunk plot.expression_macro" width="100%"></p>
<div class="sourceCode" id="cb11"><pre class="downlit">
<span class="fu">plot_expression_comparison</span>(<span class="st">"B-cell"</span>)
</pre></div>
<p><img src="figures/melanoma-plot.expression_b-1.png" title="plot of chunk plot.expression_b" alt="plot of chunk plot.expression_b" width="100%"></p>
<div class="sourceCode" id="cb12"><pre class="downlit">
<span class="fu">plot_expression_comparison</span>(<span class="st">"T-cell"</span>)
</pre></div>
<p><img src="figures/melanoma-plot.expression_t-1.png" title="plot of chunk plot.expression_t" alt="plot of chunk plot.expression_t" width="100%"></p>
</div>
<div id="differential-expression-analysis-of-spatial-clusters" class="section level2">
<h2 class="hasAnchor">
<a href="#differential-expression-analysis-of-spatial-clusters" class="anchor"></a>Differential expression analysis of spatial clusters</h2>
<p>We additionally performed a differential expression analysis between clusters of interest. We limited our analysis to the top 2,000 highly variable genes, and again we imputed their expression at enhanced resolution using xgboost.</p>
<div class="sourceCode" id="cb13"><pre class="downlit">
<span class="co">## Using the same 2,000 HVGs previously computed for PCA, excluding ribosomal</span>
<span class="kw">hvgs</span> <span class="op">&lt;-</span> <span class="kw">top</span>[<span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span>(<span class="st">"^RP[LS]"</span>, <span class="kw">top</span>, invert=<span class="fl">TRUE</span>)]

<span class="kw">melanoma.enhanced</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/enhanceFeatures.html">enhanceFeatures</a></span>(<span class="kw">melanoma.enhanced</span>, <span class="kw">melanoma</span>, 
                                     model=<span class="st">"xgboost"</span>,
                                     feature_names=<span class="kw">hvgs</span>,
                                     nrounds=<span class="fl">0</span>)
</pre></div>
<p>Next, we applied a standard Seurat differential expression analysis workflow to the enhanced resolution, finding general concordance among the histological annotations, cluster assignments, and differential gene expression.</p>
<div class="sourceCode" id="cb14"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="https://dplyr.tidyverse.org">dplyr</a></span>)

<span class="co">## Convert SCE to Seurat object and use BayesSpace cluster as identifier</span>
<span class="kw">sobj</span> <span class="op">&lt;-</span> <span class="kw">Seurat</span>::<span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/CreateSeuratObject.html">CreateSeuratObject</a></span>(counts=<span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/assays.html">logcounts</a></span>(<span class="kw">melanoma.enhanced</span>),
                                   assay=<span class="st">'Spatial'</span>,
                                   meta.data=<span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(<span class="fu">colData</span>(<span class="kw">melanoma.enhanced</span>)))
<span class="kw">sobj</span> <span class="op">&lt;-</span> <span class="kw">Seurat</span>::<span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/Idents.html">SetIdent</a></span>(<span class="kw">sobj</span>, value = <span class="st">"spatial.cluster"</span>)

<span class="co">## Scale data</span>
<span class="kw">sobj</span><span class="op">@</span>assays<span class="op">$</span><span class="kw">Spatial</span><span class="op">@</span>scale.data <span class="op">&lt;-</span>
  <span class="kw">sobj</span><span class="op">@</span>assays<span class="op">$</span><span class="kw">Spatial</span><span class="op">@</span>data <span class="op">%&gt;%</span> <span class="kw">as.matrix</span> <span class="op">%&gt;%</span> <span class="kw">t</span> <span class="op">%&gt;%</span> <span class="kw">scale</span> <span class="op">%&gt;%</span> <span class="kw">t</span>

<span class="co">## Select top n markers from each cluster (by log fold change)</span>
<span class="kw">top_markers</span> <span class="op">&lt;-</span> <span class="kw">Seurat</span>::<span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/FindAllMarkers.html">FindAllMarkers</a></span>(<span class="kw">sobj</span>, assay=<span class="st">'Spatial'</span>, slot=<span class="st">'data'</span>,
                                      group.by=<span class="st">'spatial.cluster'</span>,
                                      only.pos=<span class="fl">TRUE</span>) <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span>(<span class="kw">cluster</span>) <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/top_n.html">top_n</a></span>(<span class="fl">5</span>, <span class="kw">avg_logFC</span>)

<span class="co">## Plot expression of markers</span>
<span class="kw">Seurat</span>::<span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/DoHeatmap.html">DoHeatmap</a></span>(<span class="kw">sobj</span>, features = <span class="kw">top_markers</span><span class="op">$</span><span class="kw">gene</span>, slot=<span class="st">'scale.data'</span>,
                  group.by = <span class="st">"spatial.cluster"</span>, group.colors=<span class="kw">palette</span>, 
                  angle=<span class="fl">0</span>, size=<span class="fl">4</span>, label = <span class="fl">FALSE</span>, raster=<span class="fl">FALSE</span>) <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html">guides</a></span>(col = <span class="fl">FALSE</span>)
</pre></div>
<p><img src="figures/melanoma-seurat.DE-1.png" title="plot of chunk seurat.DE" alt="plot of chunk seurat.DE" width="100%"></p>
</div>
<div id="comparison-to-other-clustering-algorithms" class="section level2">
<h2 class="hasAnchor">
<a href="#comparison-to-other-clustering-algorithms" class="anchor"></a>Comparison to other clustering algorithms</h2>
<p>We compared the BayesSpace clustering algorithm to several other clustering algorithms, some spatially informed and some non-spatial algorithms popularly used in the analysis of single-cell data. Consistently with our earlier analyses, in all cases we cluster the first seven principal components, specifying 4 clusters when necessary.</p>
<p>We provide the code necessary to replicate the cluster assignments obtained with mclust, k-means, and the Louvain method below. For brevity, we provide the code used to run Giotto separately (adapted from their <a href="http://spatialgiotto.rc.fas.harvard.edu/giotto.visium.brain.html">Visium tutorial</a>). We could not run stLearn on this particular sample as it does not include an image aligned to the expression data.</p>
<div class="sourceCode" id="cb15"><pre class="downlit">
<span class="kw">Y1.2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html">reducedDim</a></span>(<span class="kw">melanoma</span>, <span class="st">"PCA"</span>)[, <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span>(<span class="kw">d</span>)]

<span class="co">## mclust (BayesSpace initialization)</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="https://mclust-org.github.io/mclust">mclust</a></span>)
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="fl">100</span>)
<span class="kw">mclust.labels</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mclust-org.github.io/mclust/reference/Mclust.html">Mclust</a></span>(<span class="kw">Y1.2</span>, <span class="kw">q</span>, <span class="st">"EEE"</span>)<span class="op">$</span><span class="kw">classification</span>

<span class="co">## K-means</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="fl">103</span>)
<span class="kw">km.labels</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/kmeans.html">kmeans</a></span>(<span class="kw">Y1.2</span>, centers = <span class="kw">q</span>)<span class="op">$</span><span class="kw">cluster</span>

<span class="co">## Louvain</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="fl">100</span>)
<span class="kw">g.jaccard</span> <span class="op">&lt;-</span> <span class="kw">scran</span>::<span class="fu"><a href="https://rdrr.io/pkg/scran/man/buildSNNGraph.html">buildSNNGraph</a></span>(<span class="kw">melanoma</span>, use.dimred=<span class="st">"PCA"</span>, type=<span class="st">"jaccard"</span>)
<span class="kw">louvain.labels</span> <span class="op">&lt;-</span> <span class="kw">igraph</span>::<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/cluster_louvain.html">cluster_louvain</a></span>(<span class="kw">g.jaccard</span>)<span class="op">$</span><span class="kw">membership</span>

<span class="co">## Giotto (pre-computed)</span>
<span class="kw">giotto.fname</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"2018_thrane_melanoma"</span>, <span class="st">"ST_mel1_rep2.Giotto_HMRF.csv"</span>, package = <span class="st">"BayesSpace"</span>)
<span class="kw">giotto.labels</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span>(<span class="kw">giotto.fname</span>)<span class="op">$</span><span class="kw">HMRF_PCA_k4_b.2</span>
</pre></div>
<p><img src="figures/melanoma-cluster.comparison-1.png" title="plot of chunk cluster.comparison" alt="plot of chunk cluster.comparison" width="100%"></p>
</div>
<div id="differential-expression-of-tumor-border-and-tumor-proximal-lymphoid-tissue" class="section level2">
<h2 class="hasAnchor">
<a href="#differential-expression-of-tumor-border-and-tumor-proximal-lymphoid-tissue" class="anchor"></a>Differential expression of tumor border and tumor-proximal lymphoid tissue</h2>
<p>Finally, we performed differential expression analysis between spots in the tumor border (cluster 1, in purple, above) and the spots in the lymphoid cluster (cluster 4, in yellow, above) which appeared proximal to the tumor. We highlight the spots included in the analysis below.</p>
<div class="sourceCode" id="cb16"><pre class="downlit">
<span class="kw">label_spot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Vectorize.html">Vectorize</a></span>(<span class="fu">function</span>(<span class="kw">cluster</span>, <span class="kw">col</span>) {
  <span class="co">if</span> (<span class="kw">cluster</span> <span class="op">==</span> <span class="fl">1</span> <span class="op">&amp;&amp;</span> <span class="kw">col</span> <span class="op">&lt;</span> <span class="fl">19</span>) {
    <span class="st">"Tumor border"</span>
  } <span class="co">else</span> <span class="co">if</span> (<span class="kw">cluster</span> <span class="op">==</span> <span class="fl">4</span> <span class="op">&amp;&amp;</span> <span class="kw">col</span> <span class="op">&lt;</span> <span class="fl">19</span>) {
    <span class="st">"Lymphoid"</span>
  } <span class="co">else</span> {
    <span class="st">"Other"</span>
  }
})

<span class="kw">DE.labels</span> <span class="op">&lt;-</span> <span class="fu">label_spot</span>(<span class="kw">melanoma.enhanced</span><span class="op">$</span><span class="kw">spatial.cluster</span>, <span class="kw">melanoma.enhanced</span><span class="op">$</span><span class="kw">col</span>)
<span class="kw">DE.labels</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span>(<span class="kw">DE.labels</span>, levels = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Lymphoid"</span>, <span class="st">"Tumor border"</span>, <span class="st">"Other"</span>))
<span class="fu"><a href="../reference/clusterPlot.html">clusterPlot</a></span>(<span class="kw">melanoma.enhanced</span>, label=<span class="kw">DE.labels</span>, color=<span class="st">"black"</span>, size=<span class="fl">0.1</span>) <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_fill_manual</a></span>(values=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'#0173b2'</span>, <span class="st">'#de8f05'</span>, <span class="st">'#949494'</span>)) <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span>(fill=<span class="st">"Region"</span>)
</pre></div>
<p><img src="figures/melanoma-label.tumor_border-1.png" title="plot of chunk label.tumor_border" alt="plot of chunk label.tumor_border" width="100%"></p>
<p>And perform the DE analysis as with the full set of clusters above.</p>
<div class="sourceCode" id="cb17"><pre class="downlit">
<span class="co">## Add tumor proximal/border labels</span>
<span class="kw">sobj</span> <span class="op">&lt;-</span> <span class="kw">Seurat</span>::<span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/AddMetaData.html">AddMetaData</a></span>(<span class="kw">sobj</span>, <span class="kw">DE.labels</span>, col.name=<span class="st">"DE.label"</span>)
<span class="kw">sobj</span> <span class="op">&lt;-</span> <span class="kw">Seurat</span>::<span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/Idents.html">SetIdent</a></span>(<span class="kw">sobj</span>, value = <span class="st">"DE.label"</span>)

<span class="co">## Subset to the two clusters of interest and re-scale</span>
<span class="kw">sobj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span>(<span class="kw">sobj</span>, <span class="kw">DE.label</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Lymphoid"</span>, <span class="st">"Tumor border"</span>))
<span class="kw">sobj</span><span class="op">@</span>assays<span class="op">$</span><span class="kw">Spatial</span><span class="op">@</span>scale.data <span class="op">&lt;-</span>
  <span class="kw">sobj</span><span class="op">@</span>assays<span class="op">$</span><span class="kw">Spatial</span><span class="op">@</span>data <span class="op">%&gt;%</span> <span class="kw">as.matrix</span> <span class="op">%&gt;%</span> <span class="kw">t</span> <span class="op">%&gt;%</span> <span class="kw">scale</span> <span class="op">%&gt;%</span> <span class="kw">t</span>

<span class="co">## Select top n markers from each cluster (by log fold change)</span>
<span class="kw">top_markers</span> <span class="op">&lt;-</span> <span class="kw">Seurat</span>::<span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/FindAllMarkers.html">FindAllMarkers</a></span>(<span class="kw">sobj</span>, assay=<span class="st">'Spatial'</span>, slot=<span class="st">'data'</span>,
                                      group.by=<span class="st">'DE.label'</span>,
                                      only.pos=<span class="fl">TRUE</span>) <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span>(<span class="kw">cluster</span>) <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/top_n.html">top_n</a></span>(<span class="fl">5</span>, <span class="kw">avg_logFC</span>)

<span class="co">## Plot expression of markers</span>
<span class="kw">Seurat</span>::<span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/DoHeatmap.html">DoHeatmap</a></span>(<span class="kw">sobj</span>, features = <span class="kw">top_markers</span><span class="op">$</span><span class="kw">gene</span>, slot=<span class="st">'scale.data'</span>,
                  group.by = <span class="st">"DE.label"</span>, group.colors=<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'#0173b2'</span>, <span class="st">'#de8f05'</span>), 
                  angle=<span class="fl">0</span>, size=<span class="fl">4</span>, label = <span class="fl">FALSE</span>, raster=<span class="fl">FALSE</span>) <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html">guides</a></span>(col = <span class="fl">FALSE</span>)
</pre></div>
<p><img src="figures/melanoma-tumor.DE-1.png" title="plot of chunk tumor.DE" alt="plot of chunk tumor.DE" width="100%"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Edward Zhao, Matt Stone.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
